website/src/docs/hotchocolate/performance/persisted-queries.md

```diff
@@ -2,12 +2,193 @@
+
+Example: `0c95d31ca29272475bf837f944f4e513.graphql`
+
+This file is expected to contain the query the hash was generated from.
+
+> ⚠️ Note: Do not forget to ensure that the server has access to the directory.
+
+### Redis
+
+To load persisted queries from Redis, we have to add the following package.
+
+```bash
+dotnet add package HotChocolate.PersistedQueries.Redis
+```
+
+After this we need to specify where the persisted queries are located. Using `AddReadOnlyRedisQueryStorage()` we can point to a specific Redis database in which the persisted queries are stored.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline()
+        .AddReadOnlyRedisQueryStorage(services =>
+            ConnectionMultiplexer.Connect("host:port").GetDatabase());
+}
+```
+
+TODO: Is the query hash the key? What is expected?
+
+## Hashing algorithms
+
+Per default Hot Chocolate uses the MD5 hashing algorithm, but we can override this default by specifying a `DocumentHashProvider`.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        // choose one of the following providers
+        .AddMD5DocumentHashProvider()
+        .AddSha256DocumentHashProvider()
+        .AddSha1DocumentHashProvider()
+
+        // GraphQL server configuration
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline()
+        .AddReadOnlyFileSystemQueryStorage("./persisted_queries");
+}
+```
+
+We can also configure how these hashes are encoded, by specifying a `HashFormat` as argument:
+
+```csharp
+AddSha256DocumentHashProvider(HashFormat.Hex)
+AddSha256DocumentHashProvider(HashFormat.Base64)
+```
+
+> Note: [Relay](https://relay.dev) uses the MD5 hashing algorithm - no additional Hot Chocolate configuration is needed.
+
+# Blocking regular queries
+
+We can block regular queries and only allow the execution of persisted queries, by implementing a custom request middleware.
+
+```csharp
+public class RequestMiddleware
+{
+    private readonly HotChocolate.Execution.RequestDelegate _next;
+
+    public RequestMiddleware(HotChocolate.Execution.RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public async Task InvokeAsync(IRequestContext context)
+    {
+        if (!context.IsPersistedDocument)
+        {
+            throw new GraphQLRequestException("Not a persisted document");
+        }
+
+        await _next(context);
+    }
+}
+```
+
+We need to register this middleware like the following.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddGraphQLServer()
+        .UsePersistedQueryPipeline()
+        // ...
+        .UseRequest<BlockRegularQueriesMiddleware>();
+}
+```
```

Comment:
@michaelstaib Is this what you wanted to be added for the "locking down queries" part? Is there a better way for this?